www/include/Administration/myAccount/formMyAccount.php

```diff
@@ -129,44 +129,93 @@
-        if ($centreon->user->access->page($topo2["topology_page"]) == 1) {
-            $pages[$topo2["topology_page"]] = "  " . _($topo1["topology_name"]) . " > " . _($topo2["topology_name"]);
-        }
-        $query = "SELECT topology_name, topology_parent, topology_page FROM topology " .
-            "WHERE topology_parent = '" . $topo2["topology_page"] . "' " .
-            "AND topology_page IS NOT NULL AND topology_show = '1' ORDER BY topology_group, topology_order";
-        $DBRESULT3 = $pearDB->query($query);
-        while ($topo3 = $DBRESULT3->fetchRow()) {
-            if ($centreon->user->access->page($topo3["topology_page"]) == 1) {
-                $pages[$topo3["topology_page"]] = "  " . _($topo1["topology_name"]) . " > " .
-                    _($topo2["topology_name"]) . " > " . _($topo3["topology_name"]);
+$aclUser = $centreon->user->lcaTStr;
+$acls = [];
+if (!empty($aclUser)) {
+    $acls = array_flip(explode(',', $aclUser));
+}
+
+/**
+ * Transform [1, 2, 101, 202, 10101, 20201] to :
+ *
+ * 1
+ *   101
+ *     10101
+ * 2
+ *   202
+ *     20201
+ */
+$createTopologyTree = function (array $topologies): array {
+    ksort($topologies, \SORT_ASC);
+    $parentsLvl = [];
+
+    // Classify topologies by parents
+    foreach (array_keys($topologies) as $page) {
+        if (strlen($page) == 1) {
+            if (!array_key_exists($page, $parentsLvl)) {
+                $parentsLvl[$page] = [];
+            }
+        } elseif (strlen($page) == 3) {
+            $parent = substr($page, 0, 1);
+            if (!array_key_exists($page, $parentsLvl[$parent])) {
+                $parentsLvl[$parent][$page] = [];
             }
-            $query = "SELECT topology_name, topology_page, topology_parent FROM topology " .
-                "WHERE topology_parent = '" . $topo3["topology_page"] . "' " .
-                "AND topology_page IS NOT NULL AND topology_show = '1' ORDER BY topology_order";
-            $DBRESULT4 = $pearDB->query($query);
-            while ($topo4 = $DBRESULT4->fetchRow()) {
-                if ($centreon->user->access->page($topo4["topology_page"]) == 1) {
-                    $pages[$topo4["topology_page"]] = "  " . _($topo1["topology_name"]) . " > " .
-                        _($topo2["topology_name"]) . " > " . _($topo3["topology_name"]) . " > " .
-                        _($topo4["topology_name"]);
-                }
+        } elseif (strlen($page) == 5) {
+            $parentLvl1 = substr($page, 0, 1);
+            $parentLvl2 = substr($page, 0, 3);
+            if (!array_key_exists($page, $parentsLvl[$parentLvl1][$parentLvl2])) {
+                $parentsLvl[$parentLvl1][$parentLvl2][] = $page;
+            }
+        }
+    }
+
+    return $parentsLvl;
+};
+
+$topologies = $createTopologyTree($acls);
+
+/**
+ * Retrieve the name of all topologies available for this user
+ */
+$query = "SELECT topology_page, topology_name "
+    . "FROM topology "
+    . "WHERE topology_page IN ($aclUser)";
```

Comment:
can we add AND is_react = "0" ? Because we are going to be adding the react pages in topology as well.